client.go

```diff
@@ -0,0 +1,143 @@
+	}
+}
+
+// GetBuildMetadata returns the metadata for the given Logkeeper build ID.
+func (c *LogkeeperClient) GetBuildMetadata(ctx context.Context, buildID string) (Build, error) {
+	resp, err := c.get(ctx, fmt.Sprintf("build/%s?metadata=true", buildID))
+	if err != nil {
+		return Build{}, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return Build{}, errors.Errorf("Logkeeper returned HTTP status %d", resp.StatusCode)
+	}
+
+	var build Build
+	if err := unmarshalResponseBody(resp.Body, &build); err != nil {
+		return Build{}, err
+	}
+
+	return build, nil
+}
+
+// GetTestMetadata returns the metadata for the given Logkeeper test ID.
+func (c *LogkeeperClient) GetTestMetadata(ctx context.Context, buildID, testID string) (Test, error) {
+	resp, err := c.get(ctx, fmt.Sprintf("build/%s/test/%s?metadata=true", buildID, testID))
+	if err != nil {
+		return Test{}, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return Test{}, errors.Errorf("Logkeeper returned HTTP status %d", resp.StatusCode)
+	}
+
+	var test Test
+	if err := unmarshalResponseBody(resp.Body, &test); err != nil {
+		return Test{}, err
+	}
+
+	return test, nil
+}
+
+// StreamAllLogs returns a stream of all logs from the given Logkeeper build
+// ID. It is the responsibility of the caller to close the stream.
+func (c *LogkeeperClient) StreamAllLogs(ctx context.Context, buildID string) (io.ReadCloser, error) {
+	resp, err := c.get(ctx, fmt.Sprintf("build/%s/all?raw=true", buildID))
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("Logkeeper returned HTTP status %d", resp.StatusCode)
+	}
+
+	return resp.Body, nil
+}
+
+// StreamTestLogs returns a stream of the logs from the given Logkeeper build
+// ID and test ID. It is the responsibility of the caller to close the stream.
+func (c *LogkeeperClient) StreamTestLogs(ctx context.Context, buildID, testID string) (io.ReadCloser, error) {
+	resp, err := c.get(ctx, fmt.Sprintf("build/%s/test/%s?raw=true", buildID, testID))
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("Logkeeper returned HTTP status %d", resp.StatusCode)
+	}
+
+	return resp.Body, nil
+}
+
+func (c *LogkeeperClient) get(ctx context.Context, url string) (*http.Response, error) {
+	url = fmt.Sprintf("%s/%s", c.baseURL, url)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating Logkeeper HTTP request")
+	}
+
+	httpClient := c.httpClient
+	if httpClient == nil {
+		httpClient = utility.GetHTTPClient()
+		defer utility.PutHTTPClient(httpClient)
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "doing Logkeeper HTTP request")
+	}
+
+	return resp, nil
+}
+
+func unmarshalResponseBody(body io.ReadCloser, out interface{}) error {
+	data, err := io.ReadAll(body)
```

Comment:
it's almost the same thing, but an alternative could be to 
```go
json.NewDecoder(body).Decode(out)
```

Reply:
Nice catch, thanks!

Reply:
Done